### PR TITLE
Multi core listener

### PR DESCRIPTION
**Pull request by santosh-shaastry**
* Implemented a multi-core indexer using `worker_threads`
* Upgraded `node` to `12` for `worker_threads` support
* Local testing shows `EthereumTx` coming up all right. However `EthereumBlock` looks empty (only `id` and `date` columns are populated)

### CODE REVIEW

File: ethereum-listener/src/lib/syncBlock.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    if (withMethods !== false && get(block, 'transactions', []).length) {
+      const txs = await models.EthereumTx.bulkCreate(block.transactions)
+      await syncMethodsForBlockNumber({
+        models,
+        web3,
+        ethereum,
+        blockNumber: blockNumber,
+      })
+    }
```
Comment (srslafazan) [inline ethereum-listener/src/lib/syncBlock.js:31]: Looks like the original logic is actually wrong here -- `withMethods` should only flag the `syncMethodsForBlockNumber` invocation on and off. But we can address it separately.

File: ethereum-listener/src/lib/syncBlocks.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+module.exports = async ({
+  startBlockNumber,
+  models,
+  sequelize,
+  web3,
+}) => {
+  console.log(`Starting block number ${startBlockNumber}`)
+  try {
+    let end = startBlockNumber + 10000;
```
Comment (srslafazan) [inline ethereum-listener/src/lib/syncBlocks.js:16]: We could parameterize this to `increment`

File: ethereum-listener/src/lib/syncHandler.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  const { start, end } = await getSyncStartAndEndBlocks({ models, sequelize, web3, models })
+
+  if ( !(start <= end) ) {
+    console.warn(`bad block numbers, cannot sync: Start-> ${start}, End-> ${end}`)
+    return
+  }
+
+  try {
+    const staticPool = new StaticPool({
+      size: 4,
```
Comment (srslafazan) [inline ethereum-listener/src/lib/syncHandler.js:20]: This size should be parameterized -- we can also pass it via environment variable to have more control in deployment.

**Review by srslafazan — APPROVED** [review]
LGTM! A few of the variables would be excellent to include as parameters and environment variables.

File: ethereum-listener/src/lib/syncHandler.js (hunk 3 above)
Comment (santosh-shaastry) [reply ethereum-listener/src/lib/syncHandler.js]: Absolutely. Done.

File: ethereum-listener/src/lib/syncBlock.js (hunk 1 above)
Comment (santosh-shaastry) [reply ethereum-listener/src/lib/syncBlock.js:31]: Agree. Will file a task.

**Comment by santosh-shaastry** [comment]
> * Implemented a multi-core indexer using `worker_threads`
> * Upgraded `node` to `12` for `worker_threads` support
> * Local testing shows `EthereumTx` coming up all right. However `EthereumBlock` looks empty (only `id` and `date` columns are populated)

@srslafazan confirmed that the `EthereumBlock` information is syncing all good. So we are good to proceed with the PR.

**Review by srslafazan — APPROVED** [review]
LGTM!

**Pull request merged**